src/VisualStudio/Core/Def/Implementation/HierarchyItemToProjectIdMap.cs

```diff
@@ -57,32 +47,9 @@ public bool TryGetProjectId(IVsHierarchyItem hierarchyItem, string targetFramewo
                         return false;
                     }
 
-                    // Here we try to match the hierarchy from Solution Explorer to a hierarchy from the Roslyn project.
-                    // The canonical name of a hierarchy item must be unique _within_ an hierarchy, but since we're
-                    // examining multiple hierarchies the canonical name could be the same. Indeed this happens when two
-                    // project files are in the same folder--they both use the full path to the _folder_ as the canonical
-                    // name. To distinguish them we also examine the "regular" name, which will necessarily be different
-                    // if the two projects are in the same folder.
-                    // Note that if a project has been loaded with Lightweight Solution Load it won't even have a
-                    // hierarchy, so we need to check for null first.
                     var hierarchy = _workspace.GetHierarchy(p.Id);
 
-                    if (hierarchy != null
-                        && hierarchy.TryGetCanonicalName((uint)VSConstants.VSITEMID.Root, out var projectCanonicalName)
-                        && hierarchy.TryGetItemName((uint)VSConstants.VSITEMID.Root, out var projectName)
-                        && projectCanonicalName.Equals(nestedCanonicalName, System.StringComparison.OrdinalIgnoreCase)
-                        && projectName.Equals(nestedName))
-                    {
-                        if (targetFrameworkMoniker == null)
-                        {
-                            return true;
-                        }
-
-                        return hierarchy.TryGetTargetFrameworkMoniker((uint)VSConstants.VSITEMID.Root, out var projectTargetFrameworkMoniker)
-                            && projectTargetFrameworkMoniker.Equals(targetFrameworkMoniker);
```

Comment:
Just to confirm, none of this is needed for Legacy right?

Reply:
In the case of legacy, the only time we ever have more than one project is the web intellisense project case. So if that case isn't happening, then we'll just have exactly one project with matching IVsHIerarchy, and that'll immediately be dealt with in the first "// If we only have one candidate then no further checks are required." post-processing. And the web case is handled by that horrific hack at the end of the method.